param/parameterized.py

```diff
@@ -874,18 +876,47 @@ def _changed(cls, change):
         return not Comparator.is_equal(change.old, change.new)
 
 
-    @classmethod
-    def _call_watcher(cls, watcher, change):
+
+    def _call_watcher(self_, watcher, change):
         """
         Invoke the given the watcher appropriately given a Change object.
         """
-        if watcher.onlychanged and (not cls._changed(change)):
+        if watcher.onlychanged and (not self_._changed(change)):
             return
-        if watcher.mode == 'args':
+
+        if self_.cls.param._BATCH_WATCH:
+            self_._changes.append(change)
+            self_._watchers.append(watcher)
+        elif watcher.mode == 'args':
             watcher.fn(change)
         else:
             watcher.fn(**{change.name: change.new})
 
+
+    def _batch_call_watchers(self_):
+        """
+        Batch call a set of watchers based on the parameter value
+        settings in kwargs using the queued Change and watcher objects.
+        """
+        change_dict = OrderedDict([(c.name,c) for c in self_.cls.param._changes])
+        watcher_set = []
+        for w in self_.cls.param._watchers:
+            if w not in watcher_set:
+                watcher_set.append(w)
+
+        for watcher in watcher_set:
+            changes = [change_dict[name] for name in watcher.parameter_names if name in change_dict]
+            if watcher.mode == 'args':
+                watcher.fn(*changes)
+            else:
+                watcher.fn(**{c.name:c for c in changes})
+
+        self_.cls.param._changes = []
+        self_.cls.param._watchers = []
```

Comment:
I think the queues needs to emptied before you start dispatching since a watch callback may in turn call ``.set_param``, which would mean the queue is reused and the callback would recursively call itself.

Reply:
Good point.

Reply:
Addressed in 75325ae